### PR TITLE
Copyedit of 2020 Performance chapter

### PR DESCRIPTION
**Pull request by bazzadp**
Made the following changes:

- Fix incorrect heading levels
- Normalised Smart Quotes as per #1485 and changed "it's" to "it is"
- Corrected references to 2019 chapter
- Removal of frequent use of bold in keeping with Web Almanac style
- Corrected Sheets GIDs
- Added some links
- Change of "the amount" to "the number"
- Misc edits for readability

Still to do:
- [x] - Change black on red to white on red in charts, similar to last years charts
- [x] - Font incorrect in figures 1, 12, and 13 
- [x] - Two charts (Figures 2 and 18) are wrong in sheets embed
- [x] - Add full charts for Geo breakdown as per 2019 and set heights and widths in markdown
- [x] - Re-export fallback images in higher quality (@rviscomi it appears that only those lucky enough to have 4k and 5k screens can actually export high quality images when when using the [capture node screenshots](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Figures-Guide#how-to-take-screenshots-of-google-sheet-graphics)  😞 )
- [x] - Few comments for @thefoxis to consider
- [x] - remove "unedited" flag

Staged link: https://20201204t102628-dot-webalmanac.uk.r.appspot.com/en/2020/performance (updated since initially raising this to incorporate all the changes)

### CODE REVIEW

File: src/content/en/2020/performance.md [hunk 1]
```diff
@@ -92,30 +98,32 @@ In the chart above, we can observe that between 43% and 53% of websites have goo
...
 
-India remains the last one in the set with only 16% of good experiences. While the population of people connecting to the internet for the first time is continually growing, the mobile and desktop network speeds are [still an issue](https://www.opensignal.com/reports/2020/04/india/mobile-network-experience), with average downloads of 10Mbps for 4G, 3Mbps for 3G and below 50Mbps for desktop.
+The highest percentage of good LCP readings is mostly distributed amongst European and Asian countries with the Republic of Korea (South Korea) leading at 76% of good metric readings. South Korea is a consistent leader in mobile speeds, with an impressive 145 Mbps download [reported by Speedtest Global Index](https://www.speedtest.net/global-index) for October. Japan, Czechia, Taiwan, Germany and Belgium are also a handful of countries with reliably high mobile speeds. Australia, while leading in mobile network speeds, is let down by slow desktop connections and latency which places it at the bottom section of the ranking above. {# TODO should we split this by desktop and mobile to prove this? Or at least state that we did look at this to confirm this? #}
```
Comment (bazzadp) [inline src/content/en/2020/performance.md:108]: @thefoxis thoughts?

File: src/content/en/2020/performance.md [hunk 2]
```diff
@@ -228,58 +240,64 @@ First Contentful Paint (FCP) measures the first time at which the browser render
...
+Comparing FCP on mobile devices on a year-over-year basis, we observe fewer good experiences and more moderate and poor experiences. 75% of websites have sub-par FCP. We can speculate {# TODO: maybe add a link to images chapter or js chapter if there are findings there #} this high percentage of less than ideal FCP readings is a source of frustration and degraded user experience.
 
-It’s challenging to deduce why FCP isn’t improving, as numerous factors can delay paints, such as server latency (measured by a handful of metrics, such as TTFB (TODO: add TTFB section link) and RTT), blocking JavaScript requests, or inappropriate handling of custom fonts to name a few.
+It is challenging to deduce why FCP isn't improving, as numerous factors can delay paints, such as server latency (measured by a handful of metrics, such as [Time to First Byte (TTFB)](#time-to-first-byte-ttfb) and RTT), blocking JavaScript requests, or inappropriate handling of custom fonts to name a few. {# TODO - are those not good reasons? r do you mean it's challenging to pinpoint exactly which of these is the cause (or even something else). Consider to reword #}
```
Comment (bazzadp) [inline src/content/en/2020/performance.md:278]: @thefoxis - thoughts?

File: src/content/en/2020/performance.md [hunk 3]
```diff
@@ -228,58 +240,64 @@ First Contentful Paint (FCP) measures the first time at which the browser render
...
+While the mediocre ("needs improvement") percentage becomes smaller, the number of poorly performing FCP sites rises, which is especially pronounced at the bottom of the ranking with Latin American and South Asian regions.
+
+Again, hypothesizing on reasons for the specific rankings is difficult {# TODO hypothesizing is (relatively) easy, proving is difficult #} }, especially seeing outliers such as Australia at the lower end. Australia has one of the highest global smartphone penetration levels, one of the fastest mobile networks and relatively high average income levels. We'd easily assume it should be higher up. However, taking into account slow fixed connections, latency and lack of iOS representation in CrUX, its positioning starts making more sense. With an example like this (touched only on the surface), we can see how difficult understanding context for each of the countries would be.
```
Comment (bazzadp) [inline src/content/en/2020/performance.md:300]: @thefoxis thoughts?

File: src/content/en/2020/performance.md [hunk 4]
```diff
@@ -340,39 +360,41 @@ Likening this years’ TTFB geo readings to [2019 results](https://almanac.httpa
...
+Looking at the [mobile speeds worldwide for December 2018-November 2019](https://www.speedtest.net/insights/blog/content/images/2020/02/Ookla_Mobile-Speeds-Poster_2020.png), we can see that globally, mobile connections aren't high-speed. Those network speeds and technology standards for cellular networks (such as 5G) are not evenly distributed and affect TTFB. As an example, [see this map of networks in Nigeria](https://www.mobilecoveragemaps.com/map_ng#7/8.744/7.670)—most of the country area has 2G and 3G coverage, with little 4G range.
 
-What’s surprising is the relatively the same amount of good TTFB results between offline and 4G origins. With ServiceWorkers, we could expect some of the TTFB issues to be mitigated, but that trend is not reflected in the chart above.
+What's surprising is the relatively the same number of good TTFB results between offline and 4G origins. With service workers, we could expect some of the TTFB issues to be mitigated, but that trend is not reflected in the chart above. {# TODO - any ideas why? #}
```
Comment (bazzadp) [inline src/content/en/2020/performance.md:372]: @thefoxis thoughts?

File: src/content/en/2020/performance.md [hunk 5]
```diff
@@ -340,39 +360,41 @@ Likening this years’ TTFB geo readings to [2019 results](https://almanac.httpa
...
+The chart above showcases that Performance Observer, despite being available for a while now (and [widely supported cross-browser](https://caniuse.com/mdn-api_performanceobserver)) is only used by 6-7% of tracked sites, depending on device type. Those websites will be leveraging the low-level APIs to create custom metrics, and the PerformanceObserver API to collate them, and then potentially use it with other performance reporting tooling. Such low adoption rates might indicate the tendency to lean on predefined metrics (for example, coming from Lighthouse), but also the yet to be leveraged potential of the API.
+
+{# TODO - that's actually higher than I expected as it's a bit niched. Shjould we frame this in a more positivie light as per out Glass Hlaf Full principal (https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Authors'-Guide#principles)? Also any thoughts on why this is so high? #}
```
Comment (bazzadp) [inline src/content/en/2020/performance.md:390]: @thefoxis thoughts?

**Review by bazzadp — COMMENTED** [review]
@thefoxis I've highlighted the TODOs I need your input on

File: src/content/en/2020/performance.md (hunk 1 above)
Comment (thefoxis) [reply src/content/en/2020/performance.md:103]: @bazzadp I don't think charting this is necessary. elsewhere, we don’t have charts to prove the anecdotal per-country advice. at this point it is just an anecdote :) but you can see it in the link preceding this sentence, where AU is 5th on mobile and 60th on desktop in terms of speeds

File: src/content/en/2020/performance.md (hunk 2 above)
Comment (thefoxis) [reply src/content/en/2020/performance.md:269]: @bazzadp what I’m saying here is that it's challenging to deduce why FCP is the way it is because paints can be delayed by a multitude of factors (some listed above). I don’t think we have the data (other than slow TTFB) to decidedly say "x is making FCP bad". poor TTFB definitely affects it but it’s highly contextual

File: src/content/en/2020/performance.md (hunk 3 above)
Comment (thefoxis) [reply src/content/en/2020/performance.md:291]: @bazzadp I don't see any reason to reword this to `While hypothesizing on reasons for specific rankings is easy, backing those speculations is difficult, especially seeing outliers such as Australia at the lower end.`

also, the reason there is probably the fact that iPhones aren’t counted, which is hands-down one of the most popular mobile devices in AU.

File: src/content/en/2020/performance.md (hunk 4 above)
Comment (thefoxis) [reply src/content/en/2020/performance.md:363]: @bazzadp I tried to investigate the reasoning here, but didn't find much service worker-specific.

File: src/content/en/2020/performance.md (hunk 5 above)
Comment (thefoxis) [reply src/content/en/2020/performance.md:381]: @bazzadp sure. we can rephrase to:

`The chart above showcases that Performance Observer is used by 6-7% of tracked sites, depending on device type. Those websites will be leveraging the low-level APIs to create custom metrics, and the PerformanceObserver API to collate them, and then potentially use it with other performance reporting tooling. Such adoption rates might indicate the tendency to lean on predefined metrics (for example, coming from Lighthouse), but also are impressive for a relatively niche API.`

File: src/content/en/2020/performance.md (hunk 3 above)
Comment (bazzadp) [reply src/content/en/2020/performance.md:291]: Oh I wasn't suggesting rewording to that! Sorry that wasn't clear at all.

What I meant is you have good theories here and I feel you're doing yourself a disservice in both this and the last point by down playing this with the first phrase in these sections. The current wording (of both sections) reads like "We can't guess why this is. This is probably due to..." which kind of reads contradictory.

Instead of saying it's difficult to hypothesise about why this is (and then hypothesising about this), would it sound better to just say something like "There are a number of reasons why this could be, including..."?

File: src/content/en/2020/performance.md (hunk 4 above)
Comment (bazzadp) [reply src/content/en/2020/performance.md:363]: OK will close this TODO.

File: src/content/en/2020/performance.md (hunk 5 above)
Comment (bazzadp) [reply src/content/en/2020/performance.md:381]: I like it, will change.

One more question: are you aware of any popular RUM tools or any platforms that have these built in which might explain why 6-7% of sites are using this? It seems too high to be manually coded.

Comment (thefoxis) [reply src/content/en/2020/performance.md:381]: hmm, I'm not aware of any. seems like [speedcurve has them, to some degree](https://speedcurve.com/blog/user-timing-and-custom-metrics/), but unsure if that's the exact way you're picturing it :) we have people asking about custom metrics, but very occasionally hence it's not available in calibre (yet)

Comment (bazzadp) [reply src/content/en/2020/performance.md:381]: Yeah that's the sort of thing I was thinking of. Or CDN tools like Boomerang or mPulse which might explain this - so if you're an Akamai customer do you get those automatically? We know from last year that [Akamai are 5% of third-party](https://almanac.httparchive.org/en/2019/cdn#fig-7) so that might go some way to explaining this number.

Anyway, probably delving into the weeds too much here for a Copyedit but was curious because, as I said, that was higher than I expected for being manually added.

Will change to above wording and close this. Thanks for discussing!

File: src/content/en/2020/performance.md (hunk 3 above)
Comment (thefoxis) [reply src/content/en/2020/performance.md:291]: ahh, gotcha, I misunderstood. it's not really a writing style I have, I just was trying to mimic 2019 there :) we could rephrase this para to something along these lines:

> Again, there are several reasons negatively affecting FCP, such as poor TTFB readings, but it is difficult to prove them without the necessary context. For example, if we were to analyze specific country performance, such as Australia, we surprisingly find it at the lower end. Australia has one of the highest global smartphone penetration levels, one of the fastest mobile networks and relatively high average income levels. We'd easily assume it should be higher up. However, taking into account slow fixed connections, latency and lack of iOS representation in CrUX, its positioning starts making more sense. With an example like this (touched only on the surface), we can see how difficult understanding context for each of the countries would be.

but I'm also happy to remove those couple sentences, as at this point I'm starting to doubt how useful it is 😛

File: src/content/en/2020/performance.md (hunk 1 above)
Comment (bazzadp) [reply src/content/en/2020/performance.md:103]: No probs. Resolving.

File: src/content/en/2020/performance.md (hunk 2 above)
Comment (bazzadp) [reply src/content/en/2020/performance.md:269]: I've changed to this to just this:

> Numerous factors can delay paints, such as server latency (measured by a handful of metrics, such as [Time to First Byte (TTFB)](#time-to-first-byte-ttfb) and RTT), blocking JavaScript requests, or inappropriate handling of custom fonts to name a few.

Let me know if you'd prefer something else.

File: src/content/en/2020/performance.md (hunk 3 above)
Comment (bazzadp) [reply src/content/en/2020/performance.md:291]: Thanks. Changed.

File: src/content/en/2020/performance.md (hunk 5 above)
Comment (bazzadp) [reply src/content/en/2020/performance.md:381]: Changed to suggested text.

**Comment by bazzadp** [comment]
@thefoxis that's all the TODOs done. You happy to merge this all ready for the launch next week?

**Comment by bazzadp** [comment]
Going to merge this. @thefoxis if you want to make any more changes can you open a new PR for them?

Thanks for all your work on this - looks amazing!

**Comment by thefoxis** [comment]
@bazzadp all good from my side! thank you for all the edits 🙏
